### PR TITLE
MODCAL-57 Update error code for overlapping exceptional periods

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
**Purpose**
Update overlap error status according to UI changes

Resolve: [MODCAL-57](https://issues.folio.org/browse/MODCAL-57)

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "provides": [
     {
       "id": "calendar",
-      "version": "3.0",
+      "version": "3.1",
```
Comment (kostyantyn-kh) [inline descriptors/ModuleDescriptor-template.json]: i think this is a breaking change, response was changed from 500 to 422

Comment (OleksiiMaksymov) [reply descriptors/ModuleDescriptor-template.json]: Raise the version to v.4.0

File: src/main/java/org/folio/rest/utils/CalendarUtils.java [hunk 2]
```diff
@@ -306,10 +314,31 @@ public static OpeningPeriod mapOpeningsToOpeningPeriod(Openings openings) {
...
+      errResponse = buildErrorResponse(500, e.getMessage());
     }
+
+    return succeededFuture(errResponse);
+  }
+
+  private static Response buildErrorResponse(int status, String errMessage) {
+    return Response
+      .status(status)
+      .header(CONTENT_TYPE, TEXT_PLAIN)
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/rest/utils/CalendarUtils.java]: Please use "application/json" Content-Type for 422 error response

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/rest/utils/CalendarUtils.java]: Fix

**Pull request closed without merging**